### PR TITLE
Indexing and searching studies in elasticsearch

### PR DESCRIPTION
**Pull request by SlimaneAmar**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/dto/IndexingStatus.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.study.server.dto;
+
+/**
+ * @author Slimane amar <slimane.amar at rte-france.com
+ */
+public enum IndexingStatus {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/study/server/dto/IndexingStatus.java:12]: As indexing is done synchonously by import service, the status is now useless and should be removed?

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -38,10 +38,14 @@
 
     private final StudyService studyService;
     private final ReportService reportService;
+    private final NetworkStoreService networkStoreService;
+    private final NetworkModificationService networkModificationService;
 
-    public StudyController(StudyService studyService, ReportService reportService) {
+    public StudyController(StudyService studyService, NetworkStoreService networkStoreService, NetworkModificationService networkModificationService, ReportService reportService) {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/study/server/StudyController.java:44]: Do you forgot to add an equipment search method to the controller?

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/dto/IndexingStatus.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/dto/IndexingStatus.java:12]: yes but in daily we said that we kept it anyway

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/StudyController.java:44]: yes

File: src/main/java/org/gridsuite/study/server/dto/IndexingStatus.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/dto/IndexingStatus.java]: Removed

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/study/server/StudyController.java:44]: Done

**Review by geofjamg — APPROVED** [review]

**Pull request merged**